### PR TITLE
LUN-415: Improve keyboard handling feedback modal

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

**Review by steffenkleinle — CHANGES_REQUESTED** [review]
Not working for me:

**Comment by f1sh1918** [comment]
> Not working for me: 

looks like behavior on android differs from ios :/ gonna check that

**Comment by f1sh1918** [comment]
Updated: Now i implemented a different handling for android and ios. please test on android again

File: src/hooks/useKeyboard.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
 import { useEffect, useState } from 'react'
 import { Keyboard, KeyboardEvent } from 'react-native'
 
-// This hook listens to the keyboard and returns boolean whether keyboard is opened or closed
+// This hook listens to the keyboard and returns boolean whether keyboard is opened or closed & keyBoardHeight
```
Comment (steffenkleinle) [inline src/hooks/useKeyboard.ts:4]: Optional: I think it is pretty self explanatory so I'd remove the comment
```suggestion
```

**Review by SteffiStuffel — COMMENTED** [review]
On Android it is jumping twice, looking weird (see attached video). Also the "send" button disappears. I don't think it is intuitive, that you need to scroll.

https://user-images.githubusercontent.com/18463686/190106511-88b3bafb-8625-4044-8efb-84de00a53311.mp4

**Comment by f1sh1918** [comment]
> On Android it is jumping twice, looking weird (see attached video). Also the "send" button disappears. I don't think it is intuitive, that you need to scroll.
> 
>  LUN-415.mp4

Well i dunno what to do against the jumping to be honest. Since this modal is very flexible you will reach the point where you have to scroll. We can not just resize it. So feel free for other suggestions than scrolling

**Review by SteffiStuffel — APPROVED** [review]
Not really happy with the current solution because of the jumping, but it is better than the status quo, so fine for me if it gets merged. Maybe you can open a ticket for fixing the jumping or we wait for UI/UX or testing to complain about ^^

**Comment by f1sh1918** [comment]
> Not really happy with the current solution because of the jumping, but it is better than the status quo, so fine for me if it gets merged. Maybe you can open a ticket for fixing the jumping or we wait for UI/UX or testing to complain about ^^

@SteffiStuffel 
Well the jumping issue is related to the resizing of the height that happens after the keyboard was opened. But we cannot resize the height before because we do not know the height of the keyboard before it did show.
We can only expect a particular keyboard height and calculate the Modal container height using it.
Then use this height even if keyboard is opened or closed

**Comment by SteffiStuffel** [comment]
> 

I would appreciate most, if we just block rerendering until everything is calculated and then render it (maybe useLayoutEffect will help here), as the jumping is cause by first updating the padding and in a second step updating the height.

**Comment by f1sh1918** [comment]
> > 
> 
> I would appreciate most, if we just block rerendering until everything is calculated and then render it (maybe useLayoutEffect will help here), as the jumping is cause by first updating the padding and in a second step updating the height.

fixed the bottom positioning jumping but can not get the recalc height work before updating/rendering component. you can give it a try if you want. its not just by adding `useLayoutEffect`

**Review by charludo — APPROVED** [review]
(Tested on Android Emulator)
Very nice, works as intended. Never lost track of the cursor, and all fields + send button accessible without closing the keyboard.

Minor nitpick: at least on my rather low resolution emulator screen, the margins around the modal could be smaller to show a bit more of its content, which could be useful in some cases. For example, the bottom of the email field is just barely cut off after opening the keyboard

**Pull request merged**